rfcs/192-fr-copy-receipt-performance.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: `fr_copy_receipt_performance`
+- Start Date: 2023-08-01
+- RFC PR: [fiskaltrust/middleware#192](https://github.com/fiskaltrust/middleware/pull/192)
+- Tracking Issue: [fiskaltrust/middleware#0000](https://github.com/fiskaltrust/middleware/issues/0000)
+
+# Summary
+
+Adding a `ftJournalFRCopyReceipts` table in the FR Queue where we store copy receipts.
+
+# Motivation
+
+The current implementation of the CopyReceipt is performing badly.
+Processing a CopyReceipt takes an increasing amount of time for each previous CopyReceipt in the queue.
+This inevitably leads to a point where the CopyReceipts are not processed in time.
+
+Currently when processing a CopyReceipt we have iterate over all previous CopyReceipts in the `ftJournalFR` table, deserialize the response and check if it is using the same `cbPreviousReceiptReference`.
+We then count all of those CopyReceipts and return that as the count.
+
+# Guide-level explanation
+
+For the PosCreator they are no implementation changes needed, the CopyReceipts still work like they did before.
+
+## New local queues
+
+New Queues don't need to take anything into account.
+They will work out of the box.
+
+## Updating local queues
+
+The new table will be created on start up after updating the queue.
```

Comment:
We should probably do some checks how long this can take. Maybe we can perform this also async in the background after the queue has started and becomes available?

Reply:
Hm we could maybe do that but we'll have to wait for the population of the `ftJournalFRCopyReceipts` to be completed before processing a copy receipt... I'm not sure if that complexity is worth it